lib/net/validations.rb

```diff
@@ -22,43 +25,45 @@ def valid_mac?(mac)
       false
     end
 
-    module_function :valid_mac?
+    # validates an IPv4 address
+    def self.validate_ip(ip)
+      return false unless ip.present?
+      IPAddr.new(ip, Socket::AF_INET) rescue return false
+      true
+    end
 
-    # validates the ip address
-    def validate_ip(ip)
-      raise Error, "Invalid IP Address #{ip}" unless (ip =~ IP_REGEXP)
+    # validates an IPv4 address and raises an error
+    def self.validate_ip!(ip)
+      raise Error, "Invalid IP Address #{ip}" unless validate_ip(ip)
       ip
     end
 
-    def validate_mask(mask)
+    def self.validate_mask(mask)
```

Comment:
@ekohl: To be honest, I just copied this from Dominic's branch without giving it much thought. The diff was already way too big for my taste and I tried to ignore everything not really necessary. I have cleaned the interface up though and have added some tests. Mind to take brief look?